tiledb/sm/c_api/tiledb.cc

```diff
@@ -5761,8 +5882,24 @@ void tiledb_fragment_info_free(tiledb_fragment_info_t** fragment_info) {
 
 int32_t tiledb_fragment_info_load(
     tiledb_ctx_t* ctx, tiledb_fragment_info_t* fragment_info) {
-  return tiledb_fragment_info_load_with_key(
-      ctx, fragment_info, TILEDB_NO_ENCRYPTION, nullptr, 0);
+  if (sanity_check(ctx) == TILEDB_ERR ||
+      sanity_check(ctx, fragment_info) == TILEDB_ERR)
+    return TILEDB_ERR;
+
+  // Get config from ctx
+  tiledb::sm::Config config = ctx->ctx_->storage_manager()->config();
+
+  // Load fragment info
+  if (SAVE_ERROR_CATCH(
+          ctx,
+          fragment_info->fragment_info_->load(
+              config,
+              static_cast<tiledb::sm::EncryptionType>(TILEDB_NO_ENCRYPTION),
```

Comment:
I am not sure I understand this. This function (`tiledb_fragment_info_load`) should now work for both cases of an encrypted and non-encrypted array, with the encryption mode and key being taken from the config. Then why are we hardcoding this to no encryption here?

Reply:
The logic here is that `NO_ENCRYPTION` is the default case. The original APIs used this logic so when the APIs were first deprecated, we added in the config functionality in the internal APIs. Without encryption arguments in the C API, we call the internal API (`FragmentInfo::load(...)` with "default" parameters. That internal API then checks `if (encryption_type == EncryptionType::NO_ENCRYPTION)` and retrieves the config params accordingly. This is how Joe and I decided to do the initial refactor but if you want me to change the logic I can.

Reply:
I see, this makes sense.